functorch/functorch/experimental/cond.py

```diff
@@ -0,0 +1,105 @@
+import torch
+from torch._C import DispatchKey, DispatchKeySet, ExcludeDispatchKeyGuard
+from functorch.experimental.ops import PyOperator
+from torch.utils._pytree import tree_flatten
+from torch.fx.experimental.proxy_tensor import ProxyTensor, get_isolated_graphmodule
+import torch.utils._pytree as pytree
+
+"""
+We're going to define a `cond` operation.
+In order to do this, we need implementations for each of the dispatch keys.
+"""
+from contextlib import contextmanager
+
+# TODO(voz): Move out somewhere else once other py dispatched ops need it
+@contextmanager
+def suspend_mode(mode):
+    torch._C._set_torch_dispatch_mode(None)
+    yield
+    torch._C._set_torch_dispatch_mode(mode)
+
+
+def trace_cond(proxy_mode, func_overload, args, kwargs=None):
+    assert kwargs is None or not kwargs
+    pred, true_fn, false_fn, operands = args
+
+    if isinstance(operands, ProxyTensor):
+        operands = [operands]  # Little hack because * on a single ProxyTensor unpacks it
+    else:
+        operands = operands
+
+    true_graph = get_isolated_graphmodule(true_fn, operands, {})
+    false_graph = get_isolated_graphmodule(false_fn, operands, {})
+    true_name = "true_graph"
```

Comment:
What if there are multiple conditionals in a graph? Should these be mangled so that each cond has a unique name?

Reply:
My dynamo impl handles this, and there's a comment around this below `# Does this need random slug appended so as not to collide?`. Its untested atm.

Reply:
It seems like here is the appropriate location for allocating a fresh name though